charts/gardener/resource-manager/charts/application/templates/mutatingwebhookconfiguration.yaml

```diff
@@ -291,3 +291,37 @@ webhooks:
   sideEffects: None
   timeoutSeconds: 10
 {{- end }}
+{{- if .Values.global.config.webhooks.endpointSliceHints.enabled }}
+- admissionReviewVersions:
+  - v1beta1
+  - v1
+  clientConfig:
+    {{- if .Values.global.config.server.webhooks.ca }}
+    caBundle: {{ b64enc .Values.global.config.server.webhooks.ca }}
+    {{- end }}
+    service:
+      name: gardener-resource-manager
+      namespace: {{ .Release.Namespace }}
+      path: /webhooks/endpoint-slice-hints
+      port: 443
+  failurePolicy: Ignore
```

Comment:
This should be `Fail`, right? As a user, I would expect this webhook to either work or return an error when I explicitly enable it.

Reply:
> `failurePolicy` defines how unrecognized errors and timeout errors from the admission webhook are handled. Allowed values are `Ignore` or `Fail`.
>
> `Ignore` means that an error calling the webhook is ignored and the API request is allowed to continue.
> `Fail` means that an error calling the webhook causes the admission to fail and the API request to be rejected.

I used `Ignore` because the whole topic started as cost optimization topic. I used `Ignore` because if the kube-apiserver for some reason fails to call the webhook, this to do not fail the EndpointSlice CREATE/UPDATE request. Having the EndpointSlice not created/updated is likely to cause more severe issues rather than having the EndpointSlice created/updated without the hints. Hint could be set on the next create/update request.

PS. I used `failurePolicy: Ignore` for similar reasons in the `pod-scheduler-name.resources.gardener.cloud` webhook. It more important to have the Pod created (with the `default-scheduler`) than failing the create request because the webhook for the `bin-packing-scheduler` cannot be reached.

Should I switch to `Fail`?

Reply:
> The same applies for the `schedulerName` mutation.

For the `schedulerName` mutation, see the context from https://github.com/gardener/gardener/pull/6610#issue-1358698290:

> There are 2 possible ways to tackle the issue:
> 1. Add the `default-scheduler` to the kube-scheduler explicitly when the bin-packing is enabled. This would mean if there is something wrong with the webhook and request to it fail, the Pods will be scheduled by the `default scheduler`.
> 2. Change the webhook's `failurePolicy` to `Fail` and do not allow Pods to be scheduled when the requests to the webhook fail.
>
> I picked approach 1. as it seems more reasonable to do not block the scheduling. Anyways failed requests to the webhook should be rather a rare case. This is also how I think GKE handle the webhook and kube-scheduler for their `optimize-utilization` autoscaling profile - they run the webhook with `failurePolicy: Ignore` and the `default-scheduler` seems to be present in the kube-scheduler component config.

Having in mind that the analogous webhook in GKE is with `failurePolicy: Ignore`, I would rather vote to keep the `schedulerName` webhook with `failurePolicy: Ignore`. As Shoot owner using the bin-packing scheduler, when there are issues between kube-apiserver and GRM, for me it is more important to have the Pods being created with any scheduler instead of my CREATE Pod requests being rejected.

Reply:
I now switched the `failurePolicy` of the `endpoint-slice-hints.resources.gardener.cloud` webook to `Fail`.